packages/js/components/src/form/form.tsx

```diff
@@ -268,35 +297,59 @@ function FormComponent< Values extends Record< string, any > >(
 	};
 
 	function getInputProps< Value = Values[ keyof Values ] >(
-		name: string
-	): {
-		value: Value;
-		checked: boolean;
-		selected?: boolean;
-		onChange: (
-			value: ChangeEvent< HTMLInputElement > | Values[ keyof Values ]
-		) => void;
-		onBlur: () => void;
-		className: string | undefined;
-		help: string | null | undefined;
-	} {
+		name: string,
+		inputProps: ConsumerInputProps< Values > = {}
+	): InputProps< Values, Value > {
 		const inputValue = _get( values, name );
 		const isTouched = touched[ name ];
 		const inputError = _get( errors, name );
+		const {
+			className: classNameProp,
+			onBlur: onBlurProp,
+			onChange: onChangeProp,
+			sanitize,
+			...additionalProps
+		} = inputProps;
 
 		return {
 			value: inputValue,
 			checked: Boolean( inputValue ),
 			selected: inputValue,
 			onChange: (
 				value: ChangeEvent< HTMLInputElement > | Values[ keyof Values ]
-			) => handleChange( name, value ),
-			onBlur: () => handleBlur( name ),
-			className: isTouched && inputError ? 'has-error' : undefined,
+			) => {
+				handleChange( name, value );
+				if ( onChangeProp ) {
+					onChangeProp( value );
+				}
+			},
+			onBlur: () => {
+				if ( sanitize ) {
+					handleChange( name, sanitize( inputValue ) );
+				}
+				handleBlur( name );
+				if ( onBlurProp ) {
+					onBlurProp();
+				}
+			},
+			className: classnames( classNameProp, {
+				'has-error': isTouched && inputError,
+			} ),
 			help: isTouched ? ( inputError as string ) : null,
+			...additionalProps,
 		};
 	}
 
+	function getCheckboxProps< Value = Values[ keyof Values ] >(
```

Comment:
Should this be `getCheckboxInputProps` or `getCheckboxControlProps` instead?  The reason I ask is we'll probably want to introduce a `getSelectProps` which sounds odd and could be interpreted the wrong way.  Also okay leaving it as is, but wanted to get some feedback.

Reply:
I would lean towards `getCheckboxControlProps`. But I'm also ok leaving it as it is now.

Reply:
I think that makes sense too since we're really creating props specifically for our controls 👍